internal/api/genapi/templates.go

```diff
@@ -24,7 +24,11 @@ func getArgsAndPaths(in []string, parentTypeName, action string) (colArg, resArg
 		strToReplace = in[len(in)-2]
 	}
 	colArg = fmt.Sprintf("%sId", strcase.ToLowerCamel(strings.ReplaceAll(strToReplace, "-", "_")))
-	colPath = fmt.Sprintf("%ss", in[len(in)-1])
+	colPath = in[len(in)-1]
+	// append s at the end only if it isn't already present
+	if strings.LastIndex(colPath, "s") != len(colPath)-1 {
```

Comment:
Seems strange to use a search we can just compare the last char in the string.  
```
if colPath[len(colPath)-1] != 's' {...}
```
If we don't have a guarantee the `in[len(in)-1]` is not empty we can add `len(colPath) > 0` in the front

Reply:
Thanks.